source/dub/internal/utf.d

```diff
@@ -0,0 +1,234 @@
+/**
+ * Converts binary data read from a file to UTF-8 strings.
+ *
+ * This code originates from DMD.
+ * See_Also:
+ * https://github.com/dlang/dmd/blob/a2865d74fbf43b7a4aa6a3b2db658a667cf85dc0/compiler/src/dmd/dmodule.d#L1327-L1514
+ */
+module dub.internal.utf;
+
+import std.array;
+import std.exception;
+import std.format;
+
+/**
+ * Process the content of a text file
+ *
+ * Attempts to find which encoding it is using, if it has BOM,
+ * and then normalize the text to UTF-8. If no encoding is required,
+ * a slice of `src` will be returned without extra allocation.
+ *
+ * Params:
+ *  src = Content of the text file to process
+ *  assumeASCII = Assume the the first char must be ASCII (true by default)
+ *
+ * Returns:
+ *   UTF-8 encoded variant of `src`, stripped of any BOM,
+ *   or `null` if an error happened.
+ */
+package(dub) string processText (immutable(ubyte)[] src, bool assumeASCII = true)
```

Comment:
why do we have our own implementation / copy-paste from dmd instead of using std.utf or std.encoding from phobos?

Reply:
`std.utf` doesn't deal with BOM, but I didn't think of `std.encoding`. I'll see if I can use it.